File: cpp/hot100/75.前K个高频元素.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        std::unordered_map <int, int> map;
        for(auto num : nums){
            ++map[num];
        }
        int maxFreq = nums.size();
        std::vector<std::vector<int>> buckets(maxFreq + 1);
        for(auto const& [num, freq] : map){
            buckets[freq].push_back(num);
        }
        std::vector<int> result;
        for(int i = maxFreq; i >= 0 && result.size() < k; --i){
            for(int num : buckets[i]){
                result.push_back(num);
                if(result.size() == k){
                    return result;
                }
            }
        }
        return result;
    }
};

File: python/hot100/75.前K个高频元素.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        map = collections.Counter(nums)
        max_freq = len(nums)
        buckets = [[] for  _ in range(max_freq + 1)]
        for num, frep in map.items():
            buckets[frep].append(num)
        result = []
        for i in range(max_freq, 0, -1):
            for num in buckets[i]:
                result.append(num)
                if len(result) == k:
                    return result
        return result